### PR TITLE
Add support for Picture In Picture for Element Call

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Add support for picture in picture mode for Element Call Activity

## Motivation and context

Allow user to keep seeing a call, while navigating in the Element X application or any application installed on the phone.

## Tests

- Start a call on a device > API 26
- Press back, or home button
- observe that the call is not stopped and is now displayed in PiP

Note that nothing is sent to the Element Call instance, and so the rendering of the Element Call application is not always optimal. sometimes it renders the other party video (which is fine), sometimes it render the local video loopback, which is not ideal.

## Tested devices

- [ ] Physical
- [x] Emulator API 33 and 34.
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/CallScreenView.kt [hunk 1]
```diff
@@ -58,25 +61,36 @@ interface CallScreenNavigator {
...
-            TopAppBar(
-                title = { Text(stringResource(R.string.element_call)) },
-                navigationIcon = {
-                    BackButton(
-                        imageVector = CompoundIcons.Close(),
-                        onClick = { state.eventSink(CallScreenEvents.Hangup) }
-                    )
-                }
-            )
+            if (!pipState.isInPictureInPicture) {
```
Comment (jmartinesp) [inline features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/CallScreenView.kt:79]: Technically you could also pass the web view's coordinates to the PiP code instead so it only renders that part of the activity so the webview wouldn't need a resize. It's more work, so I'm not sure it's really worth it.

File: features/preferences/impl/src/test/kotlin/io/element/android/features/preferences/impl/blockedusers/BlockedUserViewTest.kt [hunk 2]
```diff
@@ -45,7 +45,7 @@ class BlockedUserViewTest {
     fun `clicking on back invokes back callback`() {
         val eventsRecorder = EventsRecorder<BlockedUsersEvents>(expectEvents = false)
         ensureCalledOnce { callback ->
-            rule.setLogoutView(
+            rule.setBlockedUsersView(
```
Comment (jmartinesp) [inline features/preferences/impl/src/test/kotlin/io/element/android/features/preferences/impl/blockedusers/BlockedUserViewTest.kt:48]: 🙃

**Review by jmartinesp — APPROVED** [review]
LGTM, thanks! I just found a minor issue with the way we handle closing the PiP window, but it may be working as expected.

**Comment by bmarty** [comment]
Test is failing on CI, but working locally:

image

Commenting out test for version 24 in https://github.com/element-hq/element-x-android/pull/3159/commits/962320a0d8dd9c89e2e4a1e365402d5303f6892d

**Pull request merged**